infrastructure/terraform/components/dl/module_lambda_apim_key_generation.tf

```
@@ -0,0 +1,78 @@
+module "lambda_apim_key_generation" {
+  source = "https://github.com/NHSDigital/nhs-notify-shared-modules/releases/download/v2.0.20/terraform-lambda.zip"
```

I think because you've copied this from Nudge we're on an old version of the shared modules. I believe we're using `v2.0.24` for the rest of the resources in this repo:
```suggestion
  source = "https://github.com/NHSDigital/nhs-notify-shared-modules/releases/download/v2.0.24/terraform-lambda.zip"
```